backend/e4h-services/ingestion-service/app/utils/facility_service_client.py

```
@@ -39,43 +39,60 @@ def search_facility(
         hfr_id: Optional[str] = None,
         nin_id: Optional[str] = None,
     ) -> Dict[str, Any]:
-        limit = 1000
-        offset = 0
-        all_facilities = []
-
+        """
+        Search facilities with two optimized paths:
+        - ID-based (facility_id / hfr_id / nin_id): single, non-paginated call – fast for template generation.
```

Ruff (0.15.5)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)